File: web/documentserver-example/ruby/app/format/format_tests.rb
#  Copyright Ascensio System SIA 2024
#  SPDX-FileCopyrightText: 2024 Ascensio System SIA
#
#  SPDX-License-Identifier: Ascensio-System
#
#     Our License onlyoffice.com
#     Empty line
#     Empty line
#     Empty line
#     Empty line
#     Empty line
#     Empty line
#     Empty line
#     

# frozen_string_literal: true
# typed: true

require 'json'
require 'test/unit'
require_relative 'format'

# Test case for the Format class.
class FormatTests < Test::Unit::TestCase
  def test_generates_extension
    content =
      '
      {
        "name": "djvu",
        "type": "word",
        "actions": ["view"],
        "convert": ["bmp", "gif", "jpg", "pdf", "pdfa", "png"],
        "mime": ["image/vnd.djvu"]
      }
      '
    hash = JSON.parse(content)
    format = Format.from_hash(hash)
    assert_equal('.djvu', format.extension)
  end
end

# Test case for the FormatManager class, checks availability "all" formats.
class FormatManagerAllTests < Test::Unit::TestCase
  def test_loads
    format_manager = FormatManager.new
    assert_false(format_manager.all.empty?)
  end
end

# Test case for the FormatManager class, checks availability "documents" formats.
class FormatManagerDocumentsTests < Test::Unit::TestCase
  def test_loads
    format_manager = FormatManager.new
    assert_false(format_manager.documents.empty?)
  end
end

# Test case for the FormatManager class, checks availability "presentations" formats.
class FormatManagerPresentationsTests < Test::Unit::TestCase
  def test_loads
    format_manager = FormatManager.new
    assert_false(format_manager.presentations.empty?)
  end
end

# Test case for the FormatManager class, checks availability "spreadsheets" formats.
class FormatManagerSpreadsheetsTests < Test::Unit::TestCase
  def test_loads
    format_manager = FormatManager.new
    assert_false(format_manager.spreadsheets.empty?)
  end
end

# Test case for the FormatManager class, checks availability "all convertible" formats.
class FormatManagerConvertibleTests < Test::Unit::TestCase
  def test_loads
    format_manager = FormatManager.new
    assert_false(format_manager.all.empty?)
  end
end

# Test case for the FormatManager class, checks availability "all editable" formats.
class FormatManagerEditableTests < Test::Unit::TestCase
  def test_loads
    format_manager = FormatManager.new
    assert_false(format_manager.all.empty?)
  end
end

# Test case for the FormatManager class, checks availability "all viewable" formats.
class FormatManagerViewableTests < Test::Unit::TestCase
  def test_loads
    format_manager = FormatManager.new
    assert_false(format_manager.all.empty?)
  end
end

# Test case for the FormatManager class, checks availability "all filable" formats.
class FormatManagerFilableTests < Test::Unit::TestCase
  def test_loads
    format_manager = FormatManager.new
    assert_false(format_manager.all.empty?)
  end
end

File: web/documentserver-example/python/src/format/format_tests.py
#  Copyright Ascensio System SIA 2024
#  SPDX-FileCopyrightText: 2024 Ascensio System SIA
#
#  SPDX-License-Identifier: Ascensio-System
#
#     Our License onlyoffice.com
#     Empty line
#     Empty line
#     Empty line
#     Empty line
#     Empty line
#     Empty line
#     Empty line
#     

from __future__ import annotations
from unittest import TestCase
from msgspec.json import decode
from . import Format, FormatManager


class FormatTests(TestCase):
    json = \
        '''
        {
          "name": "djvu",
          "type": "word",
          "actions": ["view"],
          "convert": ["bmp", "gif", "jpg", "pdf", "pdfa", "png"],
          "mime": ["image/vnd.djvu"]
        }
        '''

    def test_generates_extension(self):
        form = decode(self.json, type=Format)
        self.assertEqual(form.extension(), '.djvu')


class FormatManagerAllTests(TestCase):
    def test_loads(self):
        format_manager = FormatManager()
        formats = format_manager.all()
        empty = len(formats) == 0
        self.assertFalse(empty)


class FormatManagerDocumentsTests(TestCase):
    def test_loads(self):
        format_manager = FormatManager()
        formats = format_manager.documents()
        empty = len(formats) == 0
        self.assertFalse(empty)


class FormatManagerPresentationsTests(TestCase):
    def test_loads(self):
        format_manager = FormatManager()
        formats = format_manager.presentations()
        empty = len(formats) == 0
        self.assertFalse(empty)


class FormatManagerSpreadsheetsTests(TestCase):
    def test_loads(self):
        format_manager = FormatManager()
        formats = format_manager.spreadsheets()
        empty = len(formats) == 0
        self.assertFalse(empty)


class FormatManagerConvertibleTests(TestCase):
    def test_loads(self):
        format_manager = FormatManager()
        formats = format_manager.convertible()
        empty = len(formats) == 0
        self.assertFalse(empty)


class FormatManagerEditableTests(TestCase):
    def test_loads(self):
        format_manager = FormatManager()
        formats = format_manager.editable()
        empty = len(formats) == 0
        self.assertFalse(empty)


class FormatManagerViewableTests(TestCase):
    def test_loads(self):
        format_manager = FormatManager()
        formats = format_manager.viewable()
        empty = len(formats) == 0
        self.assertFalse(empty)


class FormatManagerFillableTests(TestCase):
    def test_loads(self):
        format_manager = FormatManager()
        formats = format_manager.fillable()
        empty = len(formats) == 0
        self.assertFalse(empty)
